src/CanvasRenderingContext2d.cc

```diff
@@ -1206,54 +1220,29 @@ NAN_METHOD(Context2d::DrawImage) {
       //        implementation, and its not immediately clear why an offset is necessary, but without it, the result
       //        in chrome is different.
       cairo_set_source_surface(ctx, shadow_surface,
-        dx - sx + (context->state->shadowOffsetX / fx) - pad + 1.4,
-        dy - sy + (context->state->shadowOffsetY / fy) - pad + 1.4);
+        dx + context->state->shadowOffsetX - pad + 1.4,
+        dy + context->state->shadowOffsetY - pad + 1.4);
       cairo_paint(ctx);
-
       // cleanup
       cairo_destroy(shadow_context);
       cairo_surface_destroy(shadow_surface);
     } else {
       context->setSourceRGBA(context->state->shadow);
       cairo_mask_surface(ctx, surface,
-        dx - sx + (context->state->shadowOffsetX / fx),
-        dy - sy + (context->state->shadowOffsetY / fy));
+        dx + (context->state->shadowOffsetX),
+        dy + (context->state->shadowOffsetY));
     }
   }
 
-  bool sameCanvas = surface == context->canvas()->surface();
-  cairo_surface_t *surfTemp = NULL;
-  cairo_t *ctxTemp = NULL;
-
-  if (sameCanvas) {
-    int width = context->canvas()->getWidth();
-    int height = context->canvas()->getHeight();
-
-    surfTemp = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, width, height);
-    ctxTemp = cairo_create(surfTemp);
-
-    cairo_set_source_surface(ctxTemp, surface, 0, 0);
-    cairo_pattern_set_filter(cairo_get_source(ctxTemp), context->state->patternQuality);
-    cairo_pattern_set_extend(cairo_get_source(ctxTemp), CAIRO_EXTEND_REFLECT);
-    cairo_paint_with_alpha(ctxTemp, 1);
-
-    surface = surfTemp;
-  }
-
-  context->savePath();
-  cairo_rectangle(ctx, dx, dy, dw, dh);
-  cairo_clip(ctx);
-  context->restorePath();
-
   // Paint
-  cairo_set_source_surface(ctx, surface, dx - sx, dy - sy);
+  cairo_set_source_surface(ctx, surface, dx, dy);
   cairo_pattern_set_filter(cairo_get_source(ctx), context->state->patternQuality);
-  cairo_pattern_set_extend(cairo_get_source(ctx), CAIRO_EXTEND_REFLECT);
+  cairo_pattern_set_extend(cairo_get_source(ctx), CAIRO_EXTEND_NONE);
```

Comment:
Ditto.

Reply:
This has to be NONE, the previous if fine with REFLECT